src/main/java/Storage.java

```diff
@@ -2,27 +2,39 @@
 import java.io.FileNotFoundException;
 import java.io.FileWriter;
 import java.io.IOException;
+import java.util.ArrayList;
 import java.util.Scanner;
 
 public class Storage {
 
-    public static String DATA_FOLDER_PATH = "../data";
-    public static String DUKE_TXT_PATH = "../data/duke.txt";
+    public String folderPath;
+    public String filePath;
 
-    public static void makeDirectory (String filePath) throws IOException {
+    public Storage(String filePath, String folderPath) {
+        this.filePath = filePath;
+        this.folderPath = folderPath;
+
+    }
+
+    public void makeDirectory (String filePath) throws IOException {
         File file = new File(filePath);
         file.mkdirs();
     }
 
-    public static void writeToFile(String filePath, String textToAdd) throws IOException {
+    public void writeToFile(String filePath, String textToAdd) throws IOException {
         FileWriter fw = new FileWriter(filePath);
         fw.write(textToAdd);
         fw.close();
     }
 
-    public void retrieve () throws FileNotFoundException {
-        File f = new File(Storage.DUKE_TXT_PATH); // create a File for the given file path
+    public ArrayList<Task> retrieve () throws FileNotFoundException {
```

Comment:
Would it be better if you checked if the file existed, then returned an empty list, rather than throw a `FileNotFoundException` if it didn't already exist?